src/plugins/intel_gpu/src/plugin/transformations/unsqueeze_broadcast_reshape_matmul_fusion.cpp

```diff
@@ -25,118 +26,63 @@ UnsqueezeBroadcastReshapeMatmulFusion::UnsqueezeBroadcastReshapeMatmulFusion() {
+    auto unsqueeze_predicate = [](const ov::Output<ov::Node>& output) -> bool {
+        const auto unsqueeze = ov::as_type_ptr<const ov::op::v0::Unsqueeze>(output.get_node_shared_ptr());
+        if (!unsqueeze)
+            return false;
+        const auto kvcache = ov::as_type_ptr<const ov::intel_gpu::op::KVCache>(unsqueeze->get_input_node_shared_ptr(0));
+        if (!kvcache)
+            return false;
+        return rank_equals(5)(output) && consumers_count(1);
     };
 
-    auto reshape_rank_equals_and_has_static_dim = [](const ov::Output<ov::Node>& output) -> bool {
-        return rank_equals(4)(output) && has_static_dim(2) && consumers_count(1);
+    auto broadcast_predicate = [](const ov::Output<ov::Node>& output) -> bool {
+        return rank_equals(5)(output) && consumers_count(1);
     };
 
-    auto input_a_m = any_input(not_reshape);
-    auto input_b_m = any_input(not_reshape);
-
-    auto unsqueeze_a_axes_m = wrap_type<ov::op::v0::Constant>();
-    auto unsqueeze_a_m = wrap_type<ov::op::v0::Unsqueeze>({input_a_m, unsqueeze_a_axes_m}, consumers_count(1));
-    auto unsqueeze_b_axes_m = wrap_type<ov::op::v0::Constant>();
-    auto unsqueeze_b_m = wrap_type<ov::op::v0::Unsqueeze>({input_b_m, unsqueeze_b_axes_m}, consumers_count(1));
-
-    auto broadcast_a_target_shape_m = wrap_type<ov::op::v0::Constant>();
-    auto broadcast_a_m = wrap_type<ov::op::v3::Broadcast>({unsqueeze_a_m, broadcast_a_target_shape_m}, broadcast_rank_equals_and_has_static_dims);
-    auto broadcast_b_target_shape_m = wrap_type<ov::op::v0::Constant>();
-    auto broadcast_b_m = wrap_type<ov::op::v3::Broadcast>({unsqueeze_b_m, broadcast_b_target_shape_m}, broadcast_rank_equals_and_has_static_dims);
-
-    auto reshape_a_pattern_m = wrap_type<ov::op::v0::Constant>();
-    auto reshape_a_m = wrap_type<ov::op::v1::Reshape>({broadcast_a_m, reshape_a_pattern_m}, reshape_rank_equals_and_has_static_dim);
-    auto reshape_b_pattern_m = wrap_type<ov::op::v0::Constant>();
-    auto reshape_b_m = wrap_type<ov::op::v1::Reshape>({broadcast_b_m, reshape_b_pattern_m}, reshape_rank_equals_and_has_static_dim);
-
-    auto matmul_in_a = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{input_a_m, reshape_a_m});
-    auto matmul_in_b = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{input_b_m, reshape_b_m});
+    auto reshape_predicate = [](const ov::Output<ov::Node>& output) -> bool {
+        return rank_equals(4)(output) && consumers_count(1);
+    };
 
-    auto matmul_m = wrap_type<op::Gemm>({matmul_in_a, matmul_in_b});
+    auto input_a_m = any_input(not_reshape);
+    auto input_b_m = any_input();
+    auto axes_const_m = wrap_type<ov::op::v0::Constant>();
+    auto unsqueeze_m = wrap_type<ov::op::v0::Unsqueeze>({input_b_m, axes_const_m}, unsqueeze_predicate);
+    auto broadcast_m = wrap_type<ov::op::v3::Broadcast>({unsqueeze_m, any_input()}, broadcast_predicate);
+    auto pattern_const_m = wrap_type<ov::op::v0::Constant>();
```

Comment:
It it mandatory? Can it be a shape_of subgraph? I think if you remove pattern value check from callback, this can also be replaced with any_input()

Reply:
It's not mandatory. replaced it with `any_input()`